### PR TITLE
Adds Launchdarkly plugin

### PR DESCRIPTION
**Pull request by punkle**
Adds Launchdarkly plugin

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/frontend/backstage-plugin-launchdarkly/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Create an entity with the following annotations and import it:
+
+```
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: launchdarklytest
+  annotations:
+    launchdarkly.com/project-key: default
```
Comment (iain-b) [inline plugins/frontend/backstage-plugin-launchdarkly/README.md:40]: I wonder if it'd make sense to have defaults at the config level (and or component props) so we could perhaps avoid repeating these for each entity (project and particularly environment) at least some of the time.

File: plugins/frontend/backstage-plugin-launchdarkly/src/components/EntityLaunchdarklyOverviewCard/EntityLaunchdarklyOverviewCard.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  const unsetAnnotations = difference(
+    [
+      LAUNCHDARKLY_PROJECT_KEY_ANNOTATION,
+      LAUNCHDARKLY_CONTEXT_PROPERTIES_ANNOTATION,
+      LAUNCHDARKLY_ENVIRONMENT_KEY_ANNOTATION,
+    ],
+    Object.keys(entity.metadata?.annotations || {}),
+  );
+
+  const { value, error, loading } = useAsync(async () => {
```
Comment (iain-b) [inline plugins/frontend/backstage-plugin-launchdarkly/src/components/EntityLaunchdarklyOverviewCard/EntityLaunchdarklyOverviewCard.tsx:63]: I'm not sure if we plan to add another component but putting this in a hook might help reuse or even make it easier for others to make contributions.

File: plugins/frontend/backstage-plugin-launchdarkly/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          m => m.EntityLaunchdarklyOverviewCard,
+        ),
+    },
+  }),
+);
+
+export const isLaunchdarklyAvailable = (entity: Entity) => {
+  const diff = difference(
+    [
+      LAUNCHDARKLY_PROJECT_KEY_ANNOTATION,
```
Comment (iain-b) [inline plugins/frontend/backstage-plugin-launchdarkly/src/plugin.ts:52]: as above I wonder if we could make some of these optional (at least some of the time).

**Review by iain-b — COMMENTED** [review]
LGTM, one or two suggestions but I'm happy to merge as is but the build is failing (add deps to package.json) once it's fixed I'll approve.

File: plugins/frontend/backstage-plugin-launchdarkly/README.md (hunk 1 above)
Comment (punkle) [reply plugins/frontend/backstage-plugin-launchdarkly/README.md:40]: done.

File: plugins/frontend/backstage-plugin-launchdarkly/src/components/EntityLaunchdarklyOverviewCard/EntityLaunchdarklyOverviewCard.tsx (hunk 2 above)
Comment (punkle) [reply plugins/frontend/backstage-plugin-launchdarkly/src/components/EntityLaunchdarklyOverviewCard/EntityLaunchdarklyOverviewCard.tsx]: done

File: plugins/frontend/backstage-plugin-launchdarkly/src/plugin.ts (hunk 3 above)
Comment (punkle) [reply plugins/frontend/backstage-plugin-launchdarkly/src/plugin.ts:52]: done.

**Review by iain-b — APPROVED** [review]

**Pull request merged**